packages/components/src/lib/controllers/app-controller.js

```diff
@@ -892,18 +892,22 @@ export default class AppController {
       homonym.lexemes.filter((l) => l.isPopulated()).length < 1
     if (notFound && !this._store.state.app.queryStillActive) {
       let languageName
+      let languageCode
       if (homonym) {
+        languageCode = homonym.language
         languageName = this.api.app.getLanguageName(homonym.languageID).name
-      } else if (this._store.state.app.currentLanguageName) {
+      } else if (this._store.state.app.currentLanguageCode && this._store.state.app.currentLanguageName) {
+        languageCode = this._store.state.app.currentLanguageCode
         languageName = this._store.state.app.currentLanguageName
       } else {
+        languageCode = Constants.STR_LANG_CODE_UNDEFINED
         languageName = this.api.l10n.getMsg('TEXT_NOTICE_LANGUAGE_UNKNOWN')
       }
       if (this._store.state.app.lexicalRequest.source === LexicalQuery.sources.PAGE) {
         // we offer change language here when the lookup was from the page because the language used for the
         // lookup is deduced from the page and might be wrong
         const message = this.api.l10n.getMsg('TEXT_NOTICE_CHANGE_LANGUAGE',
-          { targetWord: this._store.state.app.targetWord, languageName: languageName, langCode: homonym.language })
+          { targetWord: this._store.state.app.targetWord, languageName: languageName, langCode: languageCode })
```

Comment:
Also here there is not a direct note to the current PR, but to the whole method `showLanguageInfo`.
I think that we should divide all calculations here between `homonym` instance, LexicalQuery current status and APP state properties.
1.
```
const notFound = !homonym ||
      !homonym.lexemes ||
      homonym.lexemes.length < 1 ||
      homonym.lexemes.filter((l) => l.isPopulated()).length < 1
```
This part is about homonym, defines empty homonym and I believe could be used in different places in packages. It is better to place it as a check inside `Homonym` class.

2.
```
if (this._store.state.app.currentLanguageCode && this._store.state.app.currentLanguageName) {
        languageCode = this._store.state.app.currentLanguageCode
        languageName = this._store.state.app.currentLanguageName
      } else {
        languageCode = Constants.STR_LANG_CODE_UNDEFINED
        languageName = this.api.l10n.getMsg('TEXT_NOTICE_LANGUAGE_UNKNOWN')
      }
```
This part is about defining `currentLanguageCode` and `currentLanguageName` for the current state of the sesson and is not directly related to the homonym language data. I believe that these calculations should be scoped inside definition of  `currentLanguageCode` and `currentLanguageName`.

3. 
```
if (notFound && !this._store.state.app.queryStillActive) {
if (this._store.state.app.lexicalRequest.source === LexicalQuery.sources.PAGE) {
        // we offer change language here when the lookup was from the page because the language used for the
        // lookup is deduced from the page and might be wrong
        const message = this.api.l10n.getMsg('TEXT_NOTICE_CHANGE_LANGUAGE',
          { targetWord: this._store.state.app.targetWord, languageName: languageName, langCode: languageCode })
        this._store.commit('ui/setNotification', { text: message, important: true, showLanguageSwitcher: true })
      } else {
        // if we are coming from e.g. the lookup or the wordlist, offering change language
        // here creates some confusion and the language was explicit upon lookup so it is not necessary
        const message = this.api.l10n.getMsg('TEXT_NOTICE_NOT_FOUND',
          { targetWord: this._store.state.app.targetWord, languageName: languageName, langCode: homonym ? homonym.language : '' })
        this._store.commit('ui/setNotification', { text: message, important: true, showLanguageSwitcher: false })
      }
    } else if (!this._store.state.app.queryStillActive) {
      this._store.commit('ui/resetNotification')
    }
```
And only this part is about showing ui message with languageInfo.
And here we could make this code more readable:
instead of this
```
{ targetWord: this._store.state.app.targetWord, languageName: languageName, langCode: languageCode }
```
do this (rename `languageCode ` to `langCode` or change it inside message definition)
```
const targetWord = this._store.state.app.targetWord
const message = this.api.l10n.getMsg('TEXT_NOTICE_CHANGE_LANGUAGE',
          { targetWord, languageName, languageCode })
```

I think that with applying my suggestions we could 
- reduce code amount without loosing its functionality
- reduce potential duplicate code 
- make code more readable

I believe that this my inspiration of code optimize is caused by yesterdays afforts to find out how we handle with options through a lot of long paths of code in `components` and `data-models`. May be you will find my suggestions helpful.
Anyway they don't prevent me from approving the code.